app/src/main/java/ch/admin/bag/dp3t/reports/ReportsFragment.java

```diff
@@ -214,6 +223,52 @@ private void deleteNotifications(TracingStatusInterface tracingStatusInterface)
 		builder.show();
 	}
 
+	private void setupFreeTestInfoBox(@NonNull View view) {
+
+		View testLocationsButton = view.findViewById(R.id.testlocations_link);
+		testLocationsButton.setOnClickListener(v -> showWhereToTestDialog());
+
+		TextView testCountdownTextView = view.findViewById(R.id.test_countdown_textview);
+
+		List<ExposureDay> exposureDays = tracingViewModel.getAppStatusLiveData().getValue().getExposureDays();
+		DayDate today = new DayDate();
+		DayDate oldestExposure = null; // This includes only exposures that are newer than MIN_EXPOSURE_AGE_TO_DO_A_TEST
+
+		for (ExposureDay exposureDay : exposureDays) {
+			if (exposureDay.getExposedDate().addDays(MIN_EXPOSURE_AGE_TO_DO_A_TEST).isBeforeOrEquals(today) &&
+					!exposureDay.getExposedDate().addDays(MAX_EXPOSURE_AGE_TO_DO_A_TEST).isBefore(today)) {
+				testCountdownTextView.setText(R.string.meldungen_detail_free_test_now);
+				return;
+			}
+			if (!exposureDay.getExposedDate().addDays(MIN_EXPOSURE_AGE_TO_DO_A_TEST).isBeforeOrEquals(today) &&
+					exposureDay.getExposedDate().isBeforeOrEquals(today)) {
```

Comment:
Can an `exposureDay` ever be from the future?

Reply:
No, I have removed this unnecessary check.